move the network check to the health check task

Omarabdul3ziz opened the pull request:
### Description
move the network check to the health check task

### Changes
- remove the standalone perf task network-health and add it as a check in the healthcheck task.
- modify the schedule of the healthcheck to be every 15 instead of 20 mins
- modify the report of the network check to only report the failed services for a better alignment with the other healthcheck
- updated the diagnostics package to check for the healthcheck


### Related Issues

List of related issues

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstring

Omarabdul3ziz committed:
move the network check to the health check task:

- remove the standalone perf task network-health and add it as a check
  in the healthcheck task.
- modify the schedule of the healthcheck to be every 15 instead of 20
  mins
- modify the report of the network check to only report the failed
  services for a better alignment with the other healthcheck

Omarabdul3ziz committed:
update the diagnostics manager to check for the node health instead of just the network check

muhamadazmy commented on pkg/diagnostics/diagnostics.go:
```diff
@@ -115,7 +115,7 @@ func (m *DiagnosticsManager) getModuleStatus(ctx context.Context, module string)
 }
 
-func (m *DiagnosticsManager) isOnline(ctx context.Context) bool {
+func (m *DiagnosticsManager) isHealthy() bool {
```
The idea of diagnostics to return realtime health of system modules. I am wondering if we need to return health also since health checks results can also be retrieved over the perf api which is already exposed

muhamadazmy replied:
Maybe it's better approach to have diagnostics unrelated to healthchecks. Unless there is something else i am missing

Omarabdul3ziz replied:
it was asked to add the network state for the diagnostics call for a future plan to use this call in the proxy or clients to decide if the node is reachable or not.
since we merged the checks (network/cache) i thought it is better to check for both and decide the total node health instead of just the network. 

so in the `proxy/node-indexer` the diagnostics call will be used to check for both fields `SystemOk` and `Healthy` to decide if the node is healthy or not

muhamadazmy reviewed (commented):
This is much better. But i have only one comment to discuss. Let me know what you think

Omarabdul3ziz committed:
stop sending uptime reports if the node can not reach grid services (#2215)

* stop sending uptime reports if the node can not reach grid services:

- add new `NotReachable` flag
- set the flag each time fails to resolve a service url, and unset on
      each new NetworkHealth run

* delete the notreachable flag after the run only if there is no errors
